server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -3584,9 +3600,14 @@ synchronized boolean isEmpty() {
         }
 
         synchronized boolean assertNotQueued(Snapshot snapshot) {
-            assert snapshotsToFinalize.getOrDefault(snapshot.getRepository(), new LinkedList<>())
+            if (snapshotsToFinalize.getOrDefault(snapshot.getRepository(), new LinkedList<>())
                 .stream()
-                .noneMatch(entry -> entry.equals(snapshot)) : "Snapshot [" + snapshot + "] is still in finalization queue";
+                .anyMatch(entry -> entry.equals(snapshot))) {
```

Comment:
It looks like `.stream().anyMatch(entry -> entry.equals(snapshot))` could be simplified to `.contains(snapshot)`